bundles/org.openhab.binding.bluetooth.bluegiga/src/main/java/org/openhab/binding/bluetooth/bluegiga/handler/BlueGigaBridgeHandler.java

```diff
@@ -171,92 +193,174 @@ public void handleCommand(ChannelUID channelUID, Command command) {
+                    }
+                } else {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                            "Failed opening serial port.");
+                }
+            }
+        } catch (RuntimeException e) {
+            // Avoid scheduled task to shutdown
+            // e.g. when BlueGiga module is detached
+            logger.debug("Start failed", e);
         }
+    }
 
-        final String portId = (String) getConfig().get(BlueGigaAdapterConstants.CONFIGURATION_PORT);
+    private void stop(boolean exit) {
+        if (transactionManager != null) {
+            transactionManager.removeEventListener(this);
+            transactionManager.close();
+        }
 
-        if (portId == null) {
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Serial port must be configured!");
-            return;
+        if (serialHandler != null) {
+            serialHandler.removeHandlerListener(this);
+            serialHandler.close();
         }
-        if (openSerialPort(portId, 115200)) {
-            BlueGigaSerialHandler bgh = new BlueGigaSerialHandler(inputStream, outputStream);
-            // Create and send the reset command to the dongle
-            bgh.addEventListener(this);
-            bgh.addHandlerListener(this);
-            this.setBgHandler(bgh);
-
-            updateStatus(ThingStatus.UNKNOWN);
-
-            scheduler.submit(() -> {
-                // Stop any procedures that are running
-                bgStopProcedure();
-
-                // Close all transactions
-                BlueGigaCommand command = new BlueGigaGetConnectionsCommand();
-                BlueGigaGetConnectionsResponse connectionsResponse = (BlueGigaGetConnectionsResponse) bgh
-                        .sendTransaction(command);
-                if (connectionsResponse != null) {
-                    maxConnections = connectionsResponse.getMaxconn();
-                }
 
-                // Close all connections so we start from a known position
-                for (int connection = 0; connection < maxConnections; connection++) {
-                    bgDisconnect(connection);
-                }
+        initComplete = false;
+        connections.clear();
+        closeSerialPort();
 
-                // Get our Bluetooth address
-                command = new BlueGigaAddressGetCommand();
-                BlueGigaAddressGetResponse addressResponse = (BlueGigaAddressGetResponse) bgh.sendTransaction(command);
-                if (addressResponse != null) {
-                    address = new BluetoothAddress(addressResponse.getAddress());
-                    updateStatus(ThingStatus.ONLINE);
-                } else {
-                    updateStatus(ThingStatus.OFFLINE);
-                }
+        if (exit) {
+            stopScheduledTasks();
+            initTask.cancel(true);
+            devices.clear();
```

Comment:
I would make sure to call dispose on all the devices as well to make sure they unregister from stuff.